ERCs/eip-5732/contracts/FiftyFifty.sol

```diff
@@ -0,0 +1,176 @@
+    /// @param gameId The id of the game to cancel.
+    function cancelBet(uint256 gameId) external;
+
+    /// @notice Allows the committer to reveal their choice and end the game. Depending on if the
+    ///     guesser guessed the choice, the winner will get transfered the bet amounts.
+    /// @param gameId The id of the game to reveal.
+    /// @param choice The choice the committer chose.
+    /// @param salt The salt that was used to hash the choice and create the commitment.
+    /// @dev The committer will only ever be incentivized to call this function if they win,
+    ///     because otherwise they can will let guesser call timeout and pay gas for it.
+    function reveal(
+        uint256 gameId,
+        bool choice,
+        bytes32 salt
+    ) external;
+
+    /// @notice Allows the guesser to call timeout if the committer doesn't reveal in time.
+    /// @param gameId The id of the game to timeout.
+    function timeoutBet(uint256 gameId) external;
+
+    /// @notice Returns the total number of games.
+    /// @return count The total number of games.
+    function getGameCount() external view returns (uint256 count);
+
+    /// @notice Returns the game with the given ID.
+    /// @param gameId The ID of the game to get.
+    /// @return game The game with the given ID.
+    function getGame(uint256 gameId) external view returns (Game memory game);
+}
+
+contract FiftyFifty is IFiftyFifty {
+    uint256 private gameCount = 0;
+    mapping(uint256 => Game) private games;
+
+    /// @inheritdoc IFiftyFifty
+    function commit(bytes32 _commitment) external payable {
+        require(msg.value > 0, "cannot bet with 0 value");
+
+        gameCount++;
+
+        games[gameCount].committer = payable(msg.sender);
+        games[gameCount].commitment = _commitment;
+        games[gameCount].bet = msg.value;
+
+        emit BetPlaced(gameCount, msg.value, msg.sender, _commitment);
+    }
+
+    /// @inheritdoc IFiftyFifty
+    function cancelBet(uint256 _gameId) external {
+        require(games[_gameId].ended == false, "game already ended");
+        require(msg.sender == games[_gameId].committer, "cancellable only by committer");
+        require(games[_gameId].guesser == address(0), "cancellable only if guesser not set");
+
+        games[_gameId].ended = true;
+        games[_gameId].committer.transfer(games[_gameId].bet);
+
+        emit GameEnded(_gameId, GameEndReason.CANCEL, games[_gameId].committer);
+    }
+
+    /// @inheritdoc IFiftyFifty
+    function matchBet(uint256 _gameId, bool _choice) public payable {
+        require(games[_gameId].ended == false, "game already ended");
+        require(games[_gameId].guesser == address(0), "may only place bet if nobody else has");
+        require(msg.value == games[_gameId].bet, "bet amount does not match");
+
+        games[_gameId].guesser = payable(msg.sender);
+        games[_gameId].guess = _choice;
+        games[_gameId].expiration = block.timestamp + 7 days;
```

Comment:
This is fun. Do you want `7 days` to be configurable?

Reply:
Good idea, could probably change to be configurable as a constructor arg.